app.js

```diff
@@ -1,13 +1,37 @@
-var express = require("express");
-var app = express();
-var bodyParser = require("body-parser");
+require("dotenv").config();
+const express = require("express");
+const bodyParser = require("body-parser");
+const db = require("./db");
+const session = require("express-session");
+const MySQLStore = require("express-mysql-session")(session);
 
-// configure app to use bodyParser()
-// this will let us get the data from a POST
-app.use(bodyParser.urlencoded({ extended: true }));
+const port = process.env.PORT || 8080;
+const sessionStore = new MySQLStore({}, db.pool);
+
+const app = express();
+
+app.use(
+  session({
+    secret: process.env.SESSION_SECRET,
+    store: sessionStore,
+    resave: false,
+    saveUninitialized: false,
+    cookie: {
+      httpOnly: true,
+      maxAge: 1000 * 60 * 3,
+    },
+  })
+);
 app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
 
-var port = process.env.PORT || 8080; // set our port
+const routeProtection = (req, res, next) => {
```

Comment:
Is this used on every route? 403 if user isn't signed in?

Reply:
Every route except /login, yeah. I think those changes were part of a different branch

Reply:
Oh shoot. Wait. I'm not using it at all... I forgot to put it back in after testing.